src/Dialog/Dialog.js

```diff
@@ -14,6 +15,10 @@ const defaultProps = {
 
 class Dialog extends React.Component {
     componentDidMount() {
+        if (this.props.onBackdropClick) {
+            this.backdropClickCallback = this.onDialogClick.bind(this);
+            this.dialogRef.addEventListener('click', this.backdropClickCallback);
```

Comment:
Do not bind and add the event conditionally to a props. Always add the event, and inside the event, only do something if a prop is set.

Reply:
Updated branch with requested change.